framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -42,47 +43,199 @@ class BlockSynchronizationMechanism {
 	}
 
 	async run(receivedBlock) {
+		this.active = true;
 		try {
-			this.active = true;
 			const { connectedPeers: peers } = await this.channel.invoke(
-				'network:getUniqueOutboundConnectedPeers',
+				'network:getNetworkStatus',
 			);
 			if (!peers.length) {
 				throw new Error('Connected peers list is empty');
 			}
-			// eslint-disable-next-line no-unused-vars
+
 			const bestPeer = this._computeBestPeer(peers);
```

Comment:
I like the implementation below there, however, we need to discuss about `_computeBestPeer` with P2P team @limiaspasdaniel

Reply:
Agreed, we can have a discussion about it but this PR didn't introduce this so I would keep it for a separate meeting.